src/gtk/gftp-gtk-mac-init.c

```diff
@@ -0,0 +1,65 @@
+/*****************************************************************************/
+/*  gftp-gtk-mac-init.c - macOS GTK init corner case                         */
+/*  Copyright (C) 1998-2007 Brian Masney <masneyb@gftp.org>                  */
+/*  Copyright (C) 2016 Steven Edwards <winehacker@gmail.com>                 */
+/*                                                                           */
+/*  This program is free software; you can redistribute it and/or modify     */
+/*  it under the terms of the GNU General Public License as published by     */
+/*  the Free Software Foundation; either version 2 of the License, or        */
+/*  (at your option) any later version.                                      */
+/*                                                                           */
+/*  This program is distributed in the hope that it will be useful,          */
+/*  but WITHOUT ANY WARRANTY; without even the implied warranty of           */
+/*  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the            */
+/*  GNU General Public License for more details.                             */
+/*                                                                           */
+/*  You should have received a copy of the GNU General Public License        */
+/*  along with this program; if not, write to the Free Software              */
+/*  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111 USA      */
+/*****************************************************************************/
+
+#include "gftp-gtk.h"
+
+#ifdef __APPLE__
+#include <dlfcn.h>
+
+struct ProcessSerialNumber {
+   unsigned long highLongOfPSN;
+   unsigned long lowLongOfPSN;
+};
+typedef struct ProcessSerialNumber ProcessSerialNumber;
+typedef ProcessSerialNumber * ProcessSerialNumberPtr;
+
+typedef void (*_pGetCurrentProcess)(struct ProcessSerialNumber *psn);
+typedef void (*_pCPSEnableForegroundOperation)(struct ProcessSerialNumber *psn, int val2, int val3, int val4, int val5);
+typedef void (*_pSetFrontProcess)(struct ProcessSerialNumber *psn);
+
+void gftp_gtk_mac_foreground_init( void )
```

Comment:
I would rename this function to be gftp_gtk_platform_specific_init instead of having mac in the name. I would also take mac out of the filename as well.